docs/source/DebuggingWithGDB.rst

```diff
@@ -185,3 +181,38 @@ breakpoints, executing commands and querying for output.
 
 When you check the output (``--show-job-log``) you can see that despite
 declaring the variable as 0, ff is injected and printed instead.
+
+
+rr wrapper
+----------
+
+It uses the ``--wrapper`` avocado functionality to run matching binaries
+inside the specified script. There is an example `rr` script in
+``/usr/share/avocado/wrappers/rr.sh`` which stores the information in
+test's `output_dir/rr` as `$bin_name.$serial_id`.
+
+Let's have a look at an example with malfunctioning qemu. Run the
+malfunctioning qemu inside avocado with wrapper::
+
+    $ avocado run "/usr/local/bin/qemu-system-aarch64 -machine virt" --wrapper /usr/share/examples/wrappers/rr.sh:*qemu*
+    JOB ID     : ded737a3494360c3f3ab125e6f70725d78c0a33e
+    JOB LOG    : /home/medic/avocado/job-results/job-2016-08-05T08.57-ded737a/job.log
+    TESTS      : 1
+     (1/1) /usr/local/bin/qemu-system-aarch64 -machine virt: FAIL (36.51 s)
+    RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0
+    JOB HTML   : /home/medic/avocado/job-results/job-2016-08-05T08.57-ded737a/html/results.html
+    TESTS TIME : 36.51 s
+
+Which creates the necessary files in the results directory::
+
+    $ ls ~/avocado/job-results/job-2016-08-05T08.57-ded737a/test-results/*/data/rr
+    latest-trace  qemu-system-aarch64-0
+
+So now I can re-run the execution with forward/backward step support inside gdb (step/rstep, next/rnext, ... see `<http://rr-project.org/>`_ for details)::
+
+    $ rr replay ~/avocado/job-results/job-2016-08-05T08.57-ded737a/test-results/1-_usr_local_bin_qemu-system-aarch64\ -machine\ virt/data/rr/qemu-system-aarch64-0
+    ...
+    (rr)
+
+.. warning:: Currently you can't simply re-run programs created in the test's
```

Comment:
Again, IMHO, this adds confusion.  The regular user reading the docs will be troubled by a warning that has little to do with the issue at hand.  This is a, hopefully, conscious design decision about `workdir`.  Users using `outputdir`, such as this snippet of `gdbtest.py`, will have no troubles:

```python
        self.return99_binary_path = os.path.join(self.outputdir, 'return99')
        return99_source_path = os.path.join(self.datadir, 'return99.c')
        process.system('gcc -O0 -g %s -o %s' % (return99_source_path,
                                                self.return99_binary_path))
```